### ISSUE TITLE
http2 push support added

### ISSUE THREAD

**Issue by nosovk** [jwilder/nginx-proxy#1075]
https://www.nginx.com/blog/nginx-1-13-9-http2-server-push/
Just need to add ```http2_push_preload on;```

**Comment by rodrigoaguilera** [comment]
Does it come with any kind of overhead?

I don't think is right to load this overhead to everyone since it can be enabled per-host easily.

Should the nginx-proxy image require a minimum version of nginx to support this?

**Comment by nosovk** [comment]
@rodrigoaguilera it seems that no overhead in that option. http2 is already enabled, and it's just part of protocol. That directive should just enable support of prioritization in http2

**Comment by Nottt** [comment]
Has this been implemented?

**Comment by masad-frost** [comment]
I have a fork here with server push enabled.
Docker hub: https://hub.docker.com/repository/docker/masfrost/nginx-proxy
GitHub:  https://github.com/masad-frost/nginx-proxy
Relevant commit: https://github.com/masad-frost/nginx-proxy/commit/eb9b5933f3bcb408933eb90272ee715e793c52eb

**Comment by junderw** [comment]
Can't you already add this via per-host configuration on your own?

**Comment by masad-frost** [comment]
Sure can!

**Comment by masad-frost** [comment]
Sure can!

I mean you can do anything on your own technically, but that defeats the purpose of this image

**Comment by junderw** [comment]
Yeah, I guess adding `HTTP2_PUSH=true` to your docker-compose.yml file is easier than adding a single file with `http2_push_preload on;` and then adding a volume pointing to that file in the container at `my.domain_location`...

Ever so slightly.

Also, having it in the container's document would give more exposure to the feature so more people know about it without needing to know much about Nginx.

Unless the argument is to have it on as default, adding an option is not that huge of a gain for anyone really IMO.

In regards to the default (currently off) I have no opinion on the matter. Seems easy to implement tho. I could do it if I get some time later.

**Comment by JamBalaya56562** [comment]
For context, I'm a third party going through some long-standing open issues. It looks like this feature request may now be obsolete due to upstream changes in nginx itself. HTTP/2 Server Push (the `http2_push` / `http2_push_preload` directives referenced here) was deprecated by major browsers and then removed entirely from nginx in version 1.25.1, so the directives no longer exist. The current nginx-proxy release (v1.11.2) is built on nginx 1.31.2 (see `Dockerfile.debian` / `Dockerfile.alpine`), well past that removal, meaning `http2_push_preload on;` can no longer be configured at all. HTTP/2 itself remains enabled by default and HTTP/3 support has since been added (see the "HTTP/2 and HTTP/3" section of the docs). Given this, the request can probably no longer be implemented; if maintainers agree, this might be a candidate for closing, though I'll leave that to your judgement. It may be worth a quick confirmation by anyone still interested.
